lib/server.js

```diff
@@ -253,24 +264,26 @@ module.exports = function createServer(config, log) {
               blockReason = blockReasons.IP_BAD_REPUTATION
             }
 
+            const result = {
```

Comment:
It's because `allowWhitelisted` has the hidden and surprising side-effect of modifying `result` in place. I'm definitely not a fan of side-effects so would be happy to change this to something less hidden and surprising. A clearer function name, returning the `result` maybe. Let me know if you have other suggestions.